src-test/src/org/openbravo/test/createInvoiceFromOrder/CreateInvoiceFromOrderTest.java

```diff
@@ -0,0 +1,330 @@
+package org.openbravo.test.createInvoiceFromOrder;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+
+import java.lang.reflect.Method;
+import java.math.BigDecimal;
+import java.util.Date;
+import java.util.HashMap;
+
+import org.apache.commons.lang.mutable.MutableBoolean;
+import org.codehaus.jettison.json.JSONArray;
+import org.codehaus.jettison.json.JSONException;
+import org.codehaus.jettison.json.JSONObject;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.openbravo.advpaymentmngt.test.TestUtility;
+import org.openbravo.advpaymentmngt.test.Value;
+import org.openbravo.base.provider.OBProvider;
+import org.openbravo.base.secureApp.VariablesSecureApp;
+import org.openbravo.base.weld.test.WeldBaseTest;
+import org.openbravo.client.kernel.RequestContext;
+import org.openbravo.dal.core.OBContext;
+import org.openbravo.dal.service.OBDal;
+import org.openbravo.database.ConnectionProvider;
+import org.openbravo.erpCommon.ad_process.ConvertQuotationIntoOrder;
+import org.openbravo.erpCommon.utility.OBError;
+import org.openbravo.erpCommon.utility.OBMessageUtils;
+import org.openbravo.model.ad.system.Client;
+import org.openbravo.model.common.businesspartner.BusinessPartner;
+import org.openbravo.model.common.businesspartner.Location;
+import org.openbravo.model.common.currency.Currency;
+import org.openbravo.model.common.enterprise.DocumentType;
+import org.openbravo.model.common.enterprise.Organization;
+import org.openbravo.model.common.enterprise.Warehouse;
+import org.openbravo.model.common.order.Order;
+import org.openbravo.model.common.order.OrderLine;
+import org.openbravo.model.common.plm.Product;
+import org.openbravo.model.financialmgmt.payment.FIN_PaymentMethod;
+import org.openbravo.model.financialmgmt.payment.PaymentTerm;
+import org.openbravo.model.financialmgmt.tax.TaxRate;
+import org.openbravo.model.pricing.pricelist.PriceList;
+import org.openbravo.model.pricing.pricelist.ProductPrice;
+import org.openbravo.scheduling.ProcessBundle;
+import org.openbravo.service.db.DalConnectionProvider;
+import org.openbravo.test.base.Issue;
+import org.openbravo.test.base.TestConstants;
+
+import com.smf.jobs.ActionResult;
+import com.smf.jobs.Result;
+import com.smf.jobs.defaults.invoices.CreateFromOrder;
+
+public class CreateInvoiceFromOrderTest extends WeldBaseTest {
+
+  public static final String PAYMENT_TERM = "66BA1164A7394344BB9CD1A6ECEED05D"; // 30 days
+  public static final String PRODUCT_PRICE = "4028E6C72959682B01295B03CEE40245";
```

Comment:
Is it necessary?

Reply:
Yes, because that's where I get the product and the prices.